File: 237. Delete Node in a Linked List/deleteNode.py
# Definition for singly-linked list.
from typing import List, Optional

class ListNode:
  def __init__(self, x):
    self.val = x
    self.next = None

# helper function to generate LinkedList from array
def arrayToLinkedList(arr: Optional[List]) -> Optional[ListNode]:
  if not arr:
    return None

  dummyHead = ListNode(0)
  curr = dummyHead

  for i in range(len(arr)):
    node = ListNode(arr[i])
    curr.next = node
    curr = curr.next
  
  return dummyHead.next

class Solution:
  def deleteNode(self, node):
    """
    :type node: ListNode
    :rtype: void Do not return anything, modify node in-place instead.
    """
    node.val = node.next.val
    node.next = node.next.next


head = arrayToLinkedList([4, 5, 1, 9])
node = head.next
solution = Solution()
solution.deleteNode(node)
while head:
  print(head.val)
  head = head.next

File: 237. Delete Node in a Linked List/deleteNode.js
// Definition for singly-linked list.
class ListNode {
  constructor(val = 0, next = null) {
    this.val = val;
    this.next = next;
  }
}

// helper function to generate LinkedList from array
const arrayToLinkedList = (arr) => {
  if (!arr || !arr.length) {
    return null;
  }

  const dummyHead = new ListNode(0);
  let curr = dummyHead;

  for (let i = 0; i < arr.length; i++) {
    let node = new ListNode(arr[i]);
    curr.next = node;
    curr = curr.next;
  }

  return dummyHead.next;
}
/**
 * @param {ListNode} node
 * @return {void} Do not return anything, modify node in-place instead.
 */
const deleteNode = (node) => {
  node.val = node.next.val;
  node.next = node.next.next;
};


const head = arrayToLinkedList([4, 5, 1, 9]);
const node = head.next;
deleteNode(node);
console.log('head: ', head); // [4, 1, 9]
